src/js/controller/model.js

```diff
@@ -102,7 +102,7 @@ define([
 
                     /* falls through */
                 case events.JWPLAYER_MEDIA_META:
-                    var duration = data.duration;
+                    var duration = data.duration || 0;
```

Comment:
@hghazzi It sounds like we do not need ``` _.isNumber``` check if we are making it default to 0 when duration is ```NaN```. Also, if we do it from the provider, why do we need to do the same thing here? We should probably just do it in here or in the providers, not duplicate the same behavior.